python/nav/mibs/juniper_mib.py

```diff
@@ -155,6 +158,27 @@ def _translate_fru_status_value(oper_status):
         """
         return FRU_STATUS_MAP.get(oper_status, FRU.STATE_UNKNOWN)
 
+    @defer.inlineCallbacks
+    def get_memory_usage(self):
+        """Retrieves memory usage stats from a Juniper device.
+
+        :returns: A deferred whose result is a dict
+                  {slot_type: (used_bytes, free_bytes)}
+
+        """
+        result = dict()
+        slots = yield self.retrieve_columns(
+            [OPERATING_DESCR, OPERATING_MEM, OPERATING_BUF]
+        )
+        for row in slots.values():
+            total = row[OPERATING_MEM]
+            if total:
+                # Due to operating buffer being in percent multiply by 1000 to get bytes
+                used = row[OPERATING_BUF] * total * 10000
+                free = total - used
```

Comment:
This whole block becomes quite confusing, and the comment increases the confusion, since it doesn't match up with the actual code.

`total` is a number reported in MB for `JUNIPER-MIB`. Generally, for storage values, 1 MB = 1024\*\*2 bytes = 1048576 bytes.  I don't think short-cutting the actual formula here does the reader any favors.

Also, `free = total - used` is no good, since `total` is still a number of MB, while `free` is (almost) a number of bytes. You end up with negative numbers.